tests/lib/nested.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+execute_remote(){
+    sshpass -p $PASSWORD ssh -p $SSH_PORT -q -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no $USER@localhost "$*"
+}
+
+prepare_ssh(){
+    execute_remote "sudo adduser --extrausers --quiet --disabled-password --gecos '' test"
+    execute_remote "echo test:ubuntu | sudo chpasswd"
+    execute_remote "echo 'test ALL=(ALL) NOPASSWD:ALL' | sudo tee /etc/sudoers.d/create-user-test"
+}
+
+wait_for_ssh(){
+    retry=30
+    while ! execute_remote true; do
+        retry=$(( retry - 1 ))
+        if [ $retry -le 0 ]; then
+            echo "Timed out waiting for ssh. Aborting!"
+            exit 1
+        fi
+        sleep 10
```

Comment:
Sleeping 10 seconds at a time means potentially waiting for 10 more seconds for no greater good. This is just a machine doing work.. retrying every second, or even every half a second, should be fine, I think? Note this is the _delay between retries._ It'll still wait for longer on top of that due to ssh itself.